src/Application/Actions/Person/Get.php

```diff
@@ -124,68 +130,82 @@ protected function action(Request $request, array $args): Response
 
-        $statusesThatMayBeBankFundedLater = [
-            PaymentIntent::STATUS_REQUIRES_ACTION,
-            PaymentIntent::STATUS_REQUIRES_CONFIRMATION,
-            PaymentIntent::STATUS_REQUIRES_PAYMENT_METHOD,
-        ];
-
         $paymentIntents = $this->stripeClient->paymentIntents->all([
             'customer' => $person->stripe_customer_id,
         ]);
 
         foreach ($paymentIntents->autoPagingIterator() as $paymentIntent) {
-            /**
-             * @psalm-suppress UndefinedMagicPropertyFetch
-             *
-             * We could use the \ArrayAccess interface to metadata to avoid this issue, but that would
-             * require changing test data substantially. Previous Stripe library versions declared types that
-             * allowed this.
-             */
-            $paymentIntentIsDonorFundsTip = (
-                $paymentIntent->payment_method_types === ['customer_balance'] &&
-                $paymentIntent->metadata->campaignName === self::FUND_TIPS_CAMPAIGN_NAME
-            );
-
-            if (!$paymentIntentIsDonorFundsTip) {
+            if (! $this->isPaymentIntentDonorFundsTip($paymentIntent)) {
                 continue;
             }
 
-            $paymentIntentIsPendingDonorFundsTip = in_array(
-                $paymentIntent->status,
-                $statusesThatMayBeBankFundedLater,
-                true,
-            );
-
-            // Technically we check for recently-ish created because there's not a quick way to see if it was
-            // recently confirmed.
-            $paymentIntentIsRecentlyConfirmedDonorFundsTip =
-                $paymentIntent->status === PaymentIntent::STATUS_SUCCEEDED &&
-                $paymentIntent->created > (new \DateTimeImmutable())->modify('-10 days')->getTimestamp();
-
             $currencyCode = $paymentIntent->currency;
 
-            if ($paymentIntentIsPendingDonorFundsTip) {
+            if ($this->isPaymentIntentPendingBankTransfer($paymentIntent)) {
                 if (!isset($person->pending_tip_balance[$currencyCode])) {
                     $person->pending_tip_balance[$currencyCode] = 0;
                 }
                 $person->pending_tip_balance[$currencyCode] += $paymentIntent->amount;
             }
 
-            if ($paymentIntentIsRecentlyConfirmedDonorFundsTip) {
+            if ($this->isPaymentIntentRecentlyConfirmedDonorFundsTip($paymentIntent)) {
                 if (!isset($person->recently_confirmed_tips_total[$currencyCode])) {
                     $person->recently_confirmed_tips_total[$currencyCode] = 0;
                 }
                 $person->recently_confirmed_tips_total[$currencyCode] += $paymentIntent->amount;
             }
         }
     }
+
+
+    // Param for this and following methods is docblock only as in test the param is actually stdClass.
+    /**
+     * @param PaymentIntent $paymentIntent
+     */
+    private function isPaymentIntentRecentlyConfirmedDonorFundsTip($paymentIntent): bool
```

Comment:
```suggestion
    private function isPaymentIntentRecentlyConfirmed($paymentIntent): bool
```

I think this name better captures which aspects this new fn in particular is checking

Reply:
Committed as 711c843